agent/hcp/client/telemetry_config.go

```diff
@@ -47,9 +47,9 @@ type RefreshConfig struct {
 	RefreshInterval time.Duration
 }
 
-// MetricsEnabled returns true if metrics export is enabled, i.e. a valid metrics endpoint exists.
-func (t *TelemetryConfig) MetricsEnabled() bool {
-	return t.MetricsConfig.Endpoint != nil
+// MetricsDisabled returns true if metrics export is disabled, i.e. no valid metrics endpoint exists.
+func (t *TelemetryConfig) MetricsDisabled() bool {
+	return t.MetricsConfig.Endpoint == nil
```

Comment:
I'll be honest that I am a bit confused as to why the MetricsConfig doesn't ahve a disabled field on it? It has everything else from the agent telemetry config so it seems like we should just also add disabled to it as well? Maybe you can validate my thinking there?

Reply:
ah oops yep no this is just me being a dummy, I will just add it there!

Reply:
Okay and then that endpoint check will be the disabled flag check?

Reply:
I've just updated the PR!